src/test/java/org/prebid/server/bidder/impactify/ImpactifyBidderTest.java

```diff
@@ -0,0 +1,326 @@
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.assertj.core.api.Assertions.tuple;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.BDDMockito.given;
+
+public class ImpactifyBidderTest extends VertxTest {
+
+    private static final String TEST_ENDPOINT = "https://test.endpoint.com";
+    private static final String INCORRECT_TEST_ENDPOINT = "incorrect.endpoint";
+
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    private ImpactifyBidder impactifyBidder;
+
+    @Mock
+    private CurrencyConversionService currencyConversionService;
+
+    @Before
+    public void setUp() {
+        impactifyBidder = new ImpactifyBidder(TEST_ENDPOINT, jacksonMapper, currencyConversionService);
+    }
+
+    @Test
+    public void createBidderWithWrongEndpointShouldThrowException() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new ImpactifyBidder(INCORRECT_TEST_ENDPOINT,
+                jacksonMapper, currencyConversionService));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldConvertCurrencyIfNotDefault() {
+        // given
+        given(currencyConversionService.convertCurrency(any(), any(), anyString(), anyString()))
+                .willReturn(BigDecimal.TEN);
+
+        final BidRequest bidRequest = givenBidRequest(
+                impCustomizer -> impCustomizer.bidfloor(BigDecimal.ONE).bidfloorcur("EUR"));
+
+        // when
+        Result<List<HttpRequest<BidRequest>>> result = impactifyBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1)
+                .extracting(httpRequest -> mapper.readValue(httpRequest.getBody(), BidRequest.class))
+                .flatExtracting(BidRequest::getImp)
+                .extracting(Imp::getBidfloor, Imp::getBidfloorcur)
+                .containsExactly(tuple(BigDecimal.TEN, "USD"));
+    }
+
+    @Test
+    public void makeHttpRequestsWithValidDataWillThrowExceptionOnCurrencyConversion() {
+        // given
+        given(currencyConversionService.convertCurrency(any(), any(), anyString(), anyString()))
+                .willThrow(PreBidException.class);
+
+        final BidRequest bidRequest = givenBidRequest(
+                impCustomizer -> impCustomizer.bidfloor(BigDecimal.ONE).bidfloorcur("EUR"));
+
+        // when
+        Result<List<HttpRequest<BidRequest>>> result = impactifyBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1)
+                .extracting(BidderError::getMessage)
+                .containsExactly("Unable to convert provided bid floor currency from EUR to USD for imp `123`");
+
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnValidBidResponseWithAllHeadersExceptIpv6() {
```

Comment:
From test name exceptIpv6, in expected result `tuple(HttpUtil.X_FORWARDED_FOR_HEADER.toString(), "ipv6")`. Seems something wrong